src/library/scala/concurrent/Future.scala

```diff
@@ -604,22 +654,42 @@ object Future {
   *  @param executor  the execution context on which the future is run
   *  @return          the `Future` holding the result of the computation
   */
-  def apply[T](body: =>T)(implicit executor: ExecutionContext): Future[T] =
+  final def apply[T](body: =>T)(implicit executor: ExecutionContext): Future[T] =
     unit.map(_ => body)
 
-  /** Simple version of `Future.traverse`. Asynchronously and non-blockingly transforms a `IterableOnce[Future[A]]`
+  /** Starts an asynchronous computation and returns a `Future` instance with the result of that computation once it completes.
+  *
+  *  The following expressions are semantically equivalent:
+  *
+  *  {{{
+  *  val f1 = Future(expr).flatten
+  *  val f2 = Future.defer(expr)
+  *  val f3 = Future.unit.flatMap(_ => expr)
+  *  }}}
+  *
+  *  The result becomes available once the resulting Future of the asynchronous computation is completed.
+  *
+  *  @tparam T        the type of the result
+  *  @param body      the asynchronous computation, returning a Future
+  *  @param executor  the execution context on which the `body` is evaluated in
+  *  @return          the `Future` holding the result of the computation
+  */
+  final def defer[T](body: => Future[T])(implicit executor: ExecutionContext): Future[T] =
```

Comment:
What is a use case that motivates this method? Is it for cases in which `body` is expensive to execute? If so

  1. I think the scaladoc should have a note about that
  2. I'm concerned about encouraging such behaviour. If `body` takes a significant amount of time to execute, it should be in a `blocking` block, and probably not on the global `ExecutionContext`; however, neither of those things are indicated to be necessary by the name `defer`

To me, `defer` implies that we're waiting to create the `Future` result until later. However

 - if the `ExecutionContext`'s queue is empty, it will effectively happen immediately
 - if `body` uses the same `ExecutionContext` as `defer`, it is just throwing it on the `ExecutionContext`'s queue twice, which seems... weird and not useful

Reply:
@NthPortal The method defers execution to the given ExecutionContext. It is less about expensive, and more about controlling *where* things get executed. No matter the method, if there is blocking, then wrapping in blocking{} should be encouraged.

Reply:
`defer` doesn't seem a good name since people might think that'll postpone the execution. Maybe `fork` or `flatten`?

Reply:
@fwbrasil `flatApply`? :)

Reply:
Sounds good!

Reply:
@NthPortal @fwbrasil How about `delegate`? ("flatApply" isn't very user friendly)

Reply:
I like `delegate`

Reply:
So, will it be renamed?

Reply:
I think "defer" is too misleading, not that it's technically wrong, but people will misunderstand it anyway.

What is a concrete example use case? And why isn't `Future { makeFuture() }.flatten` good enough?

Reply:
So, will it be renamed?

Reply:
@nafg it is performance-wise more along the lines of `Future.unit.flatMap(_ => expr)` but that might be hard for people to find out.

Reply:
anyone object to `delegate`, which @viktorklang suggested a couple months ago?